docs/migration-v1-v2.md

```diff
@@ -0,0 +1,162 @@
+VPC Endpoint, or other egress by adding your own entry into the route table.
+
+
+### `subnet_type_tag_key` and `subnet_type_tag_value_format`
+The inputs `subnet_type_tag_key` and `subnet_type_tag_value_format` have
+been deprecated, and, critically, the default for `subnet_type_tag_key`
+has changed from `"cpco.io/subnet/type"` to `null`. This means that
+if you did not previously set a value for `subnet_type_tag_key`, your
+subnets would have been tagged with the `"cpco.io/subnet/type"` tag in v1.0
+and those tags will be removed in v2.0. One mitigating circumstance is
+that likely, if you were using the default value, you were not actually
+using the resulting tags for anything, so removing the tags will likely
+not affect you.
+
+The purpose of these tags was to be used as a filter to `data.aws_subnet_ids`
+so you could automatically find all the public or private subnets and not
+require them as inputs. Since there was no standard tag key for this purpose,
+we provided `subnet_type_tag_key` as an option and automatically generated
+those tags with values of either `public` or `private`. Unfortunately,
+since there was no standard tag key for this purpose, we created a default
+tag key of our own, which is less than ideal.
+
+If you were depending on these tags, then you should have been setting
+your own value for `subnet_type_tag_key`, and the good news is that if you
+continue to do so, then the tags will continue to be created as before. We do recommend you stop using
+these particular tags at some convenient point in the future. If you want to tag public and/or private
+subnets in some way to distinguish them, we suggest that instead you look at [standard
+tag keys](https://aws.amazon.com/premiumsupport/knowledge-center/eks-vpc-subnet-discovery/)
+like `kubernetes.io/role/elb` and `kubernetes.io/role/internal-elb`
+and add them to `public_subnets_additional_tags` and `private_subnets_additional_tags`
+respectively.
+
+
+### vpc_default_route_table_id -> public_route_table_ids (list)
+
+Providing a route table ID for the public subnets was always optional,
+but previously you could only supply a single route table ID to cover
+all the public subnets. We have enhanced this feature to allow you to
+supply separate route tables for each subnet, and so we have replaced
+`vpc_default_route_table_id` with the more powerful `public_route_table_ids` variable.
+
+***Migration***: Replace
+
+```hcl
+vpc_default_route_table_id = aws_vpc.main.default_route_table_id
+```
+
+with:
+
+```hcl
+public_route_table_ids = [aws_vpc.main.default_route_table_id]
+```
+
+## Removed variables
+
+### `public_network_acl_id` and `private_network_acl_id`
+
+The variables `public_network_acl_id` and `private_network_acl_id` have been removed.
+They did not work properly anyway, so it is unlikely you were using them.
+You can now control the creation of network ACLs via `public_open_network_acl_enabled`
+and `private_open_network_acl_enabled`.
```

Comment:
```suggestion
We discovered a bug in the code where supplying these variables did not behave as intended. The module now supports the creation of network ACLs via `public_open_network_acl_enabled` and `private_open_network_acl_enabled`.
```

Reply:
The module always supported the optional creation of network ACLs. What it failed to do was associate the provided ACL with the subnets.